content/en/docs/rosenpass-tool/guides/ubuntu.md

```diff
@@ -0,0 +1,275 @@
+
+<span class="codebox">
+
+```sh
+cargo build --release
+```
+
+</span>
+
+5. **Install:** If the compilation succeeded, proceed to install Rosenpass by copying its binaries to `/usr/local/bin` with root privileges:
+
+```sh{class="code-block-list"}
+sudo install target/release/rosenpass /usr/local/bin
+```
+```sh{class="code-block-list"}
+sudo install rp /usr/local/bin
+```
+<span class="spacer"></span>
+The second binary, `rp`, is the Rosenpass helper. You can skip this step if you do not want to use it.
+<span class="spacer"></span>
+
+**That's it! You have downloaded, compiled, and installed Rosenpass.** You can now proceed to [set up a Rosenpass-enhanced WireGuard VPN](#set-up-a-rosenpass-enhanced-wireguard-vpn).
+
+<span class="spacer"></span>
+
+{{% /blocks/section %}}
+
+
+{{< blocks/lead color="secondary" class="title-box" >}}
+## Set up a Rosenpass-enhanced WireGuard VPN
+{{< /blocks/lead >}}
+
+{{% blocks/section color="light" class="no-flex contains-code-snippets compilation" %}}
+
+In this section, we set up a Rosenpass-enhanced WireGuard connection between two peers. Technically, there's no difference between the two peers. However, for clarity, we name them <span class="primary-highlight"><strong>server</strong> (pink)</span> and <span class="secondary-highlight"><strong>client</strong> (orange)</span>.
+
+### Prepare the Key Pairs
+
+#### 1. Start by generating secret keys for both peers
+
+**Note:** These will be stored in newly created `server.rosenpass-secret` and `client.rosenpass-secret` directories.
+
+```sh{class="starter-code-server"}
+rp genkey server.rosenpass-secret
+```
+```sh{class="starter-code-client"}
+rp genkey client.rosenpass-secret
+```
+
+#### 2. Extract the public keys
+
+**Note:** As above, these will be stored in newly created `server.rosenpass-public` and `client.rosenpass-public directories`.
+
+```sh{class="starter-code-server"}
+rp pubkey server.rosenpass-secret server.rosenpass-public
+```
+```sh{class="starter-code-client"}
+rp pubkey client.rosenpass-secret client.rosenpass-public
+```
+
+#### 3. Copy each `-public` directory to the other peer
+
+```sh{class="starter-code-server"}
+scp -r server.rosenpass-public user@client:/path/to/directory
+```
+```sh{class="starter-code-client"}
+scp -r client.rosenpass-public user@server:/path/to/directory
+```
+
+**Congrats! This completes the setup of the key pairs.**
+
+
+### Launch your Rosenpass-enhanced WireGuard VPN
+
+#### 4. Start the VPN
+
+**Note:** This may conflict with your firewall. In that case, you will need to configure your firewall to give Rosenpass access to the port number explicitly mentioned in these commands, `9999` in this example, as well as give WireGuard access to this port number incremented by one, `10000` in this example.
+
+```sh{class="starter-code-server"}
+sudo rp exchange server.rosenpass-secret dev rosenpass0 listen 127.0.0.1:9999 \
+peer client.rosenpass-public allowed-ips fe80::/64
```

Comment:
For clarity I'd do something like this:

```sh
sudo rp exchange server.rosenpass-secret \
   dev rosenpass0 \
   listen 127.0.0.1:9999 \
   peer client.rosenpass-public \
      allowed-ips fe80::/64
```

Reply:
I like this, and also changed some other command lines accordingly.